tianshou/policy/imitation/cql.py

```diff
@@ -209,22 +209,11 @@ def learn(self, batch: Batch, **kwargs: Any) -> Dict[str, float]:
             batch_size * self.num_repeat_actions, act.shape[-1]
         ).uniform_(-self.min_action, self.max_action).to(self.device)
 
-        if len(obs.shape) > 2:
-            tmp_obs = obs.unsqueeze(1) \
-                .repeat(1, self.num_repeat_actions, 1, 1) \
-                .view(batch_size * self.num_repeat_actions,
-                      obs.shape[-2], obs.shape[-1])
-            tmp_obs_next = obs_next.unsqueeze(1) \
-                .repeat(1, self.num_repeat_actions, 1, 1) \
-                .view(batch_size * self.num_repeat_actions,
-                      obs.shape[-2], obs.shape[-1])
-        else:
-            tmp_obs = obs.unsqueeze(1) \
-                .repeat(1, self.num_repeat_actions, 1) \
-                .view(batch_size * self.num_repeat_actions, obs.shape[-1])
-            tmp_obs_next = obs_next.unsqueeze(1) \
-                .repeat(1, self.num_repeat_actions, 1) \
-                .view(batch_size * self.num_repeat_actions, obs.shape[-1])
+        obs_len = len(obs.shape)
+        repeat_size = [1, self.num_repeat_actions] + [1] * (obs_len - 1)
+        view_size = [batch_size * self.num_repeat_actions] + list(obs.shape[1:])
+        tmp_obs = obs.unsqueeze(1).repeat(*repeat_size).view(*view_size)
+        tmp_obs_next = obs_next.unsqueeze(1).repeat(*repeat_size).view(*view_size)
```

Comment:
I'm still curious about the possibility to rewrite n-step return in this style. But it's ok for me to keep the current BDQ code as is.

Reply:
Yes something to do in future for sure :)